src/commands/KickCommand.tsx

```diff
@@ -157,7 +157,10 @@ export const DraupnirKickCommand = describeCommand({
     const usersToKick: UsersToKick = new Map();
     for (const revision of roomsToKickWithin) {
       for (const member of revision.members()) {
-        if (kickRule.test(member.userID)) {
+        if (
+          kickRule.test(member.userID) &&
+          ["invite", "join", "knock"].includes(member.membership)
+        ) {
```

Comment:
```suggestion
        switch (member.membership)
          case Membership.Knock:
          case Membership.Join:
          case Membership.Invite: {
            if (kickRule.test(membership.userID)) {
              addUserToKick(
                usersToKick,
                revision.room.toRoomIDOrAlias(),
                member.userID
              );
            }
          }
```

Reply:
Something like this might be better meow, `Membership` is an enum from `matrix-protection-suite`, what dod you think?

Reply:
Oh it is? my bad, looked like a plain string to me :sweat_smile: 

This is likely more efficient anyway :+1: